registry/storage/garbagecollect.go

```diff
@@ -74,6 +75,10 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 					// which means that we need check (and delete) those references when deleting manifest
 					allTags, err := repository.Tags(ctx).All(ctx)
 					if err != nil {
+						if errors.Is(err, distribution.ErrRepositoryUnknown{Name: repository.Named().Name()}) {
```

Comment:
`repository.Named().Name()` isn't free, we're going to call it anyways to perform this err check so store it and then use it again in `emit` later, but what about the `repoName` string the `ingester` func uses above, is that not sufficient for this error check and the affiliated `emit`?

```suggestion
						if errors.Is(err, distribution.ErrRepositoryUnknown{Name: repoName}) {
```